docs/DeveloperGuide.md

```diff
@@ -1,38 +1,314 @@
+Concept wise, this treats the doOnDate as the date and time when the task occurs,
+or the date and time the user has set out to accomplish the task.
+
+
+The **Timetable** is a class which interacts with the following components:
+1. Parser
+2. ShowCommand
+3. TaskList
+
+#### Parser Component
+
+The **Parser** is a class which parses the inputs which the user enters. 
+To activate the timetable, the user inputs commands that start with `show`.
+This creates a **ShowCommand** object which will execute its method, thereby printing the timetable.
+
+#### ShowCommand Component
+
+Depending on the user input that was parsed by **Parser**, **ShowCommand** will call the 
+relative methods which prints the timetable.
+
+#### TaskList Component
+
+As **ShowCommand** is being executed, it will retrieve a filtered list
+of task by the date that is defined in the **ShowCommand** from the **TaskList** component. 
+The filtered list represents the schedule that the user has on that given date. 
+The list is assumed to be sorted previously when the user added/edited a task.
+
+Below is a sequence diagram of what happens 
+as the user wishes to see the schedule (timetable) for 25th May 2022:
+
+![](images/showScheduleForADate.png)
+
+The sequence as shown above also happens in the same fashion as the user 
+requests to see the schedule for any day or the week the user is at.
+
+The timetable for the current day is also shown to user as the user starts up
+the program.
+
+#### Design considerations for Timetable class
+- Current Implementation: Printing of timetable from scratch.
+  - Pros: Easy to implement as timetable is generated based on request and input.
+  - Pros: Adaptive as the timetable is only generated when needed and formatting is taken care of while generating it.
+  - Cons: Significant time may be taken as timetable will have to be created from scratch. The delay may be extended if user has a lot of tasks.
+
+- Alternative Implementation: Having a few templates, before choosing the suitable template and editing it if needed.
+  - Pros: Reduces computation time
+  - Cons: Increases memory usage
+
+### Storage Implementation
+#### Loading saved files
+
+Class diagram of Storage:
+
+![](images/StorageClass.png)
+
+(_Note: some methods and attributes of `TaskList`,`Ui` and `Parser` are omitted here_)
+
+The storage component
+- Can save the content of a `TaskList` to a file in JSON format
+- Can load a JSON file to restore a previously saved `TaskList`
+- Relies on the `Parser` class to understand the content of a JSON file
+- Communicates with the user through the `Ui` class
+
+The loading of a save file is done with the function
+
+`Storage#load()` - Loads a saved JSON file and returns an ArrayList of task
+
+The path of the JSON file is provided as a parameter in the constructor of `Storage` hence 
+there is no need for any parameters in the `Storage#load()`. Since a save file will be created in the 
```

Comment:
Not sure if there is a formatting error here or if it was intentional but why is there a # betwen Storage and its load() method? Is it supposed to be Storage.load() in this case?